spec/app/ics-027-interchain-accounts/README.md

```diff
@@ -113,86 +102,119 @@ function ExecuteTx(sourcePort: string, destPort: string, destChannel: string, ms
 function SetInterchainAccountAddress(portId string, address string) returns (string) {
 }
 
 // Gets the interchain account from state
-function GetInterchainAccountAddress(portId string) returns (string, error){
+function GetInterchainAccountAddress(portId string) returns (string, bool){
 }
 ```
 
-### Registering & Controlling flows
-**Active Channels**
+### Register & Controlling flows
+
+#### Register Account Flow
+
+To register an interchain account we require an off-chain process (relayer) to listen for `OnChannelOpenInit` events with the capability to finish a channel creation handshake on a given connection. 
+
+1. The controller chain binds a new IBC port with an id composed of the *source/counterparty connection-ids* & the *interchain account owner address*
+
+The IBC port identifier will look like this:
+```
+ics27-1.{connection-id}.{counterparty-connection-id}.{owner-address}
+```
+
+This port will be used to create channels between the controller & host chain for a specific owner/interchain account pair. Only the account with `{owner-address}` matching the bound port will be authorized to send IBC packets over channels created with `ics27-1.{connection-id}.{counterparty-connection-id}.{owner-address}`. It is up to each controller chain to enforce this port registration and access on the controller side. 
+
+2. The controller chain emits an event signaling to open a new channel on this port given a connection 
+3. A relayer listening for `OnChannelOpenInit` events will begin the channel creation handshake
+4. During the `OnChanOpenTry` callback on the host chain an interchain account will be registered and a mapping of the interchain account address to the owner account address will be stored in state (this is used for authenticating transactions on the host chain at execution time). 
+5. During the `OnChanOpenAck` callback on the controller chain a record of the interchain account address registered on the host chain during `OnChanOpenTry` is set in state with a mapping from portID -> interchain account address. See [version negotiation](#Version-Negotiation) section below for how to implement this
+6. During the `OnChanOpenAck` & `OnChanOpenConfirm` callbacks on the controller & host chains respectively, the [active-channel](#Active-Channels) for this interchain account/owner pair, is set in state
+
+#### Active Channels
 
 The controller and host chain should keep track of an `active-channel` for each registered interchain account. The `active-channel` is set during the channel creation handshake process. This is a safety mechanism that allows a controller chain to regain access to an interchain account on a host chain in case of a channel closing. 
 
-An active channel can look like this:
+An active channel on the controller chain will look like this:
 
 
 ```typescript
 {
  // Controller Chain
- SourcePortId: `ics-27-0-0-cosmos1mjk79fjjgpplak5wq838w0yd982gzkyfrk07am`,
+ SourcePortId: `ics27-1.0.0.<owner-id>`,
```

Comment:
Also it doesn't make sense here to give an example `connection-id` and `counterparty-connection-id` but leave the owner address as a variable.

Either give them all an example value, or leave them all as variables

Reply:
Yeah, I agree, a good point. I removed the cosmos address as it was specific to the hub but I can put it back in.

Reply:
I actually put them all as variables in the end. :handshake: